collection-scripts/common.sh

```
@@ -78,6 +83,18 @@ get_log_collection_args() {
+	# log_collection_args, node_log_collection_args, rotated_pod_logs_arg, compress_service_logs, compress_after_gather.
+	export log_collection_args node_log_collection_args rotated_pod_logs_arg compress_service_logs compress_after_gather
+}
+
+# Compress collected .log / .log.* files larger than 10MB (skip already-gzipped).
+# Uses gzip -1 (fastest). Parallelism defaults to 4; override with COMPRESS_LOGS_JOBS.
+compress_logs() {
+	local target_dir="${1:-/must-gather}"
+	local max_jobs="${COMPRESS_LOGS_JOBS:-4}"
+
+	echo "Compressing collected logs in parallel (jobs=${max_jobs})..."
+	find "${target_dir}" \( -name '*.log' -o -name '*.log.*' \) ! -name '*.gz' -size +10M \
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)